### PR TITLE
Cleanup configuration

### PR DESCRIPTION
**Pull request by mproch**

### CODE REVIEW

**Review by arkadius — APPROVED** [review]

File: engine/flink/components/kafka/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/FlinkKafkaComponentProvider.scala [hunk 1]
```diff
@@ -31,16 +33,20 @@ class FlinkKafkaComponentProvider extends ComponentProvider {
...
-      ComponentDefinition("kafka-registry-typed-json", new KafkaAvroSourceFactory(jsonSerializingSchemaRegistryProvider, overriddenDependencies, new FlinkKafkaSourceImplFactory(None))),
-      ComponentDefinition("kafka-registry-typed-json", new KafkaAvroSinkFactoryWithEditor(jsonSerializingSchemaRegistryProvider, overriddenDependencies, FlinkKafkaAvroSinkImplFactory)),
-      ComponentDefinition("kafka-registry-typed-json-raw", new KafkaAvroSinkFactory(jsonSerializingSchemaRegistryProvider, overriddenDependencies, FlinkKafkaAvroSinkImplFactory)),
-      ComponentDefinition("kafka-avro-raw", new KafkaAvroSinkFactory(avroSerializingSchemaRegistryProvider, overriddenDependencies, FlinkKafkaAvroSinkImplFactory))
+      ComponentDefinition("kafka-json", new GenericKafkaJsonSinkFactory(overriddenDependencies)).withRelativeDocs(noTypeInfo),
+      ComponentDefinition("kafka-json", new GenericJsonSourceFactory(overriddenDependencies)).withRelativeDocs(noTypeInfo),
+      ComponentDefinition("kafka-typed-json", new GenericTypedJsonSourceFactory(overriddenDependencies)).withRelativeDocs("DataSourcesAndSinks#schema-registry--json-serialization"),
```
Comment (dswiecki) [inline engine/flink/components/kafka/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/FlinkKafkaComponentProvider.scala:43]: schemaRegistryTypedJson

File: engine/flink/components/kafka/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/FlinkKafkaComponentProvider.scala [hunk 2]
```diff
@@ -31,16 +33,20 @@ class FlinkKafkaComponentProvider extends ComponentProvider {
 
   override def create(config: Config, dependencies: ProcessObjectDependencies): List[ComponentDefinition] = {
     val overriddenDependencies = TemporaryKafkaConfigMapping.prepareDependencies(config, dependencies)
+    implicit val docsConfig: DocsConfig = new DocsConfig(config)
+    val avro = "DataSourcesAndSinks#schema-registry--avro-serialization"
+    val schemaRegistryTypedJson = "DataSourcesAndSinks#schema-registry--json-serialization"
```
Comment (dswiecki) [inline engine/flink/components/kafka/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/FlinkKafkaComponentProvider.scala:38]: just json ?

File: engine/lite/components/kafka/src/main/scala/pl/touk/nussknacker/engine/lite/components/LiteKafkaComponentProvider.scala [hunk 3]
```diff
@@ -30,21 +32,28 @@ class LiteKafkaComponentProvider extends ComponentProvider {
...
+    val noTypeInfo = "DataSourcesAndSinks#no-type-information--json-serialization"
+
+    List(
+      ComponentDefinition("kafka-json", new KafkaSinkFactory(GenericJsonSerialization(_), dependencies, LiteKafkaSinkImplFactory)).withRelativeDocs(noTypeInfo),
+      ComponentDefinition("kafka-json", new KafkaSourceFactory[String, java.util.Map[_, _]](
+        ConsumerRecordDeserializationSchemaFactory.fixedValueDeserialization(deserializeToMap), jsonFormatterFactory, dependencies, new LiteKafkaSourceImplFactory)).withRelativeDocs(noTypeInfo),
+      ComponentDefinition("kafka-typed-json", new KafkaSourceFactory[String, TypedMap](
+        ConsumerRecordDeserializationSchemaFactory.fixedValueDeserialization(deserializeToTypedMap),
+        jsonFormatterFactory, dependencies, new LiteKafkaSourceImplFactory
+      ) with BaseGenericTypedJsonSourceFactory).withRelativeDocs("DataSourcesAndSinks#schema-registry--json-serialization"),
```
Comment (dswiecki) [inline engine/lite/components/kafka/src/main/scala/pl/touk/nussknacker/engine/lite/components/LiteKafkaComponentProvider.scala:48]: as above

File: engine/lite/components/kafka/src/main/scala/pl/touk/nussknacker/engine/lite/components/LiteKafkaComponentProvider.scala [hunk 4]
```diff
@@ -30,21 +32,28 @@ class LiteKafkaComponentProvider extends ComponentProvider {
...
-    ComponentDefinition("kafka-avro", new KafkaAvroSinkFactoryWithEditor(avroSerializingSchemaRegistryProvider, dependencies, LiteKafkaAvroSinkImplFactory)),
-    ComponentDefinition("kafka-registry-typed-json", new KafkaAvroSourceFactory(jsonSerializingSchemaRegistryProvider, dependencies, new LiteKafkaSourceImplFactory)),
-    ComponentDefinition("kafka-registry-typed-json", new KafkaAvroSinkFactoryWithEditor(jsonSerializingSchemaRegistryProvider, dependencies, LiteKafkaAvroSinkImplFactory)),
-    ComponentDefinition("kafka-registry-typed-json-raw", new KafkaAvroSinkFactory(jsonSerializingSchemaRegistryProvider, dependencies, LiteKafkaAvroSinkImplFactory)),
-    ComponentDefinition("kafka-avro-raw", new KafkaAvroSinkFactory(avroSerializingSchemaRegistryProvider, dependencies, LiteKafkaAvroSinkImplFactory))
-  )
+  override def create(config: Config, dependencies: ProcessObjectDependencies): List[ComponentDefinition] = {
+    implicit val docsConfig: DocsConfig = new DocsConfig(config)
+    val avro = "DataSourcesAndSinks#schema-registry--avro-serialization"
+    val schemaRegistryTypedJson = "DataSourcesAndSinks#schema-registry--json-serialization"
```
Comment (dswiecki) [inline engine/lite/components/kafka/src/main/scala/pl/touk/nussknacker/engine/lite/components/LiteKafkaComponentProvider.scala:38]: as above

**Review by dswiecki — APPROVED** [review]

File: engine/flink/components/base/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/FlinkBaseComponentProvider.scala [hunk 5]
```diff
@@ -6,30 +6,33 @@ import pl.touk.nussknacker.engine.api.process.{ProcessObjectDependencies, SinkFa
...
 
 class FlinkBaseComponentProvider extends ComponentProvider {
   override def providerName: String = "base"
 
   override def resolveConfigForExecution(config: Config): Config = config
 
   override def create(config: Config, dependencies: ProcessObjectDependencies): List[ComponentDefinition] = {
-    statefulComponents ++ statelessComponents
+    val docsConfig = new DocsConfig(config)
+    statefulComponents(docsConfig) ++ statelessComponents(docsConfig)
```
[the remark this answers is not in the document]
Comment (mproch) [reply engine/flink/components/base/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/FlinkBaseComponentProvider.scala]: ok, as you wish ;)

File: engine/flink/components/kafka/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/FlinkKafkaComponentProvider.scala (hunk 2 above)
Comment (mproch) [reply engine/flink/components/kafka/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/FlinkKafkaComponentProvider.scala:38]: fixed one link...

File: engine/lite/components/kafka/src/main/scala/pl/touk/nussknacker/engine/lite/components/LiteKafkaComponentProvider.scala (hunk 4 above)
Comment (mproch) [reply engine/lite/components/kafka/src/main/scala/pl/touk/nussknacker/engine/lite/components/LiteKafkaComponentProvider.scala:38]: fixed one link...

File: engine/lite/components/kafka/src/main/scala/pl/touk/nussknacker/engine/lite/components/LiteKafkaComponentProvider.scala (hunk 3 above)
Comment (mproch) [reply engine/lite/components/kafka/src/main/scala/pl/touk/nussknacker/engine/lite/components/LiteKafkaComponentProvider.scala]: fixed one link...

File: engine/flink/components/kafka/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/FlinkKafkaComponentProvider.scala (hunk 1 above)
Comment (mproch) [reply engine/flink/components/kafka/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/FlinkKafkaComponentProvider.scala]: fixed one link...

**Pull request merged**